src/StorageFolding.cpp

```diff
@@ -150,17 +198,12 @@ class AttemptStorageFoldingOfFunction : public IRMutator {
         }
 
         // Any folds that took place folded dimensions away entirely, so we can proceed recursively.
-        if (const For *f = result.as<For>()) {
-            Stmt body = mutate(f->body);
-            if (body.same_as(f->body)) {
-                stmt = result;
-            } else {
-                stmt = For::make(op->name, op->min, op->extent, op->for_type, op->device_api, body);
-            }
+        body = mutate(body);
+        if (body.same_as(op->body)) {
```

Comment:
this looks a bit weird, shouldn't you compare it with the prev 'body'? I mean it may be possible that it hits line 185, but on the subsequent update nothing happens.